### PR TITLE
Added support for recycling per fixed number of requests

### PR DESCRIPTION
**Pull request by AviIsraeli**
No testing was provided (noob here...) - will write them joyfully with some help :)

### CODE REVIEW

**Comment by igalic** [comment]
@AviIsraeli thank you very much for this patch a couple of remarks:

* We are currently working on a rewrite of this module: #87, that means that we need to either land this patch before #87, or we'll have to land it on a separate branch…
* if you use the [Github flow](https://guides.github.com/introduction/flow/index.html) workflow, it'll be much easier for you to incorporate our feedback
* if you need help writing tests, you can join us on IRC (freenode/#voxpupuli), and we'll gladly help you (copy paste things from [spec/defines/app_pool](https://github.com/voxpupuli/puppet-iis/blob/master/spec/defines/manage_app_pool_spec.rb);)

**Comment by AviIsraeli** [comment]
Igor,
Thank you for your response, do you have an ETA for the refactor?

Thanks,
Avi

**Comment by igalic** [comment]
ping @cyberious

**Comment by AviIsraeli** [comment]
@igalic, @cyberious 
Should I push my luck and try to insert the code to the merge_branch or should I wait for the refactor to finish and then add support for recycling per fixed number of requests? 

Thanks!

**Comment by igalic** [comment]
@AviIsraeli ping
over in #93 we've discussed that we're ready to merge
so i'd like to get this merged before, if you have the cycles
and then i wanna release this module and branch off

File: manifests/manage_app_pool.pp [hunk 1]
```diff
@@ -249,7 +251,25 @@
         logoutput => true,
       }
     }
-
+    if ($apppool_recycle_fixed_requests) {
+      if (empty($apppool_recycle_fixed_requests)) {
```
Comment (petems) [inline manifests/manage_app_pool.pp]: @AviIsraeli I'm not sure I understand the logic here: surely this should be just a number not an array? What's the usecase of setting it to an empty array?

**Comment by jyaworski** [comment]
Can this be rebased and squashed?

**Comment by igalic** [comment]
not automatically :(

**Pull request closed without merging**